tests/testthat/test-report.R

```diff
@@ -0,0 +1,88 @@
+context("report")
+
+report_test <- function() {
+    dat <- data.frame(
+        V1 = c(1, 2, 3),
+        V2 = c("a", "b", "c")
+    )
+
+    report <- data_validation_report()
+
+    validate(dat)  %>%
+        validate_cols(is.numeric)  %>%
+        validate_if(.data$V1 > 0)  %>%
+        add_results(report)
+
+    return(report)
+}
+
+get_advanced_view <- function(log) {
+    lines <- log[grep("type", log):length(log)]
+    lines <- lines[!grepl("^[[:blank:]+-=:_|]*$", lines)]
+    out <- t(
+        as.data.frame(
+            strsplit(gsub(" ", "", lines), "\\|"),
+            row.names = 1
+        )
+    )
+    out <- data.frame(out)
+
+    return(out)
+}
+
+test_that("exported CSV matched results obtained from get_results", {
+    tmp <- file.path(tempdir(), "test.csv")
+    on.exit(unlink(tmp))
+
+    report <- report_test()
+    save_results(report, tmp, method = write)
+
+    actual <- read.csv(tmp) %>%
+        dplyr::select(-"X") %>%
+        dplyr::mutate_all(as.character)
+
+    expected <- get_results(report, unnest = TRUE) %>%
+        as.data.frame() %>%
+        dplyr::mutate_all(as.character)
+
+    expect_equal(actual, expected)
+})
+
+test_that("can choose not to present success result in exported log", {
```

Comment:
That English would benefit some refactor. How about: "Success tests might be not hidden in the exported log."